kano_profile/badges.py

```diff
@@ -69,6 +69,29 @@ def calculate_xp():
     return int(points) + qm.evaluate_xp()
 
 
+def calculate_app_progress(app_name):
+    def get_group_progress(group_profile):
+        if 'challengeNo' not in group_profile:
+            return 0
+
+        return group_profile['challengeNo']
+
+    app_profile = load_app_state(app_name)
+    progress = 0
+
+    if 'groups' in app_profile and app_profile['groups']:
+        groups_profile = app_profile['groups']
+        for group in groups_profile:
+            progress += get_group_progress(groups_profile[group])
+
+        return progress
+
+    if 'level' in app_profile and app_profile['level']:
+        progress = app_profile['level']
+
+    return progress
+
+
 def calculate_kano_level():
```

Comment:
Yeah, the problem right now is that we have both the challenge groups and the `level` in use so we have to handle both cases; if we could 'convert' old profiles to new ones only using challenge groups then it would be better.